src/aap_eda/api/serializers/eda_credential.py

```diff
@@ -115,23 +115,45 @@ class EdaCredentialCreateSerializer(serializers.ModelSerializer):
     inputs = serializers.JSONField()
 
     def validate(self, data):
-        credential_type_id = data.get("credential_type_id")
-        if credential_type_id:
-            credential_type = models.CredentialType.objects.get(
-                id=credential_type_id
-            )
-        else:
-            # for update
-            credential_type = self.instance.credential_type
+        credential_type = models.CredentialType.objects.get(
+            id=data.get("credential_type_id")
+        )
 
         inputs = data.get("inputs", {})
+        errors = validate_inputs(credential_type.inputs, inputs)
+        if bool(errors):
+            raise serializers.ValidationError(errors)
 
-        # allow emtpy inputs during updating
+        return data
+
+    class Meta:
+        model = models.EdaCredential
+        fields = [
+            "name",
+            "description",
+            "inputs",
+            "credential_type_id",
+            "organization_id",
+        ]
+
+
+class EdaCredentialUpdateSerializer(serializers.ModelSerializer):
+    organization_id = serializers.IntegerField(
+        required=True,
+        allow_null=False,
+        validators=[validators.check_if_organization_exists],
+    )
```

Comment:
For an update wouldn't this be optional?

Reply:
we shouldn't allow updating the organization either, right?

Reply:
That's a good point but it still somehow works when we don't pass `organization_id` field in patch payload, I can't tell exactly how..

Reply:
```
..when using a PATCH request, fields that are required on the model or serializer are not necessarily required in the payload. This behavior is due to the partial nature of the PATCH method, which is designed to update only the fields provided in the request rather than requiring all fields.
```
Regardless, I agree that it can be confusing to set it as required. I think I could even take out the custom definition of this field since it is already in the model.

Reply:
@hsong-rh  
> we shouldn't allow updating the organization either, right?

Moving a credential from one org to another is different than changing the kind of credential it is.  Though it does raise the question of what impact it might have on RBAC?  If people who are in the old org lose the access they had at least they aren't getting access they shouldn't, but is that (and only that)  what would happen?  Might they get access to an org to which they don't belong?  Possibly dependent on what constellation of accesses are checked; if both being in the org and having the credential then that would appear fine, they'd be denied. If simply having the credential for an org that could be an issue.

Reply:
@matburt Any insights you can share or point us to someone who can?  I think there was someone from one of the staff meetings who was going to think about some of these issues but I don't remember who and no one reached out to me. 🤷‍♂️

Reply:
> @matburt Any insights you can share or point us to someone who can?  I think there was someone from one of the staff meetings who was going to think about some of these issues but I don't remember who and no one reached out to me. 🤷‍♂️ 

Actually, I think that was about the Default organization and not credential-related.

Reply:
@jshimkus-rh @hsong-rh Controller allows updating the organization for a credential; I'd say it is probably better to align with it.